analyzers/tests/SonarAnalyzer.UnitTest/Trackers/ConstantValueFinderTest.cs

```diff
@@ -57,5 +57,34 @@ public int Method()
             var finder = new CSharpConstantValueFinder(compilation.GetSemanticModel(tree));
             finder.FindConstant(returnExpression).Should().Be(42);
         }
+
+        [TestMethod]
+        public void WrongCompilationBeingUsed()
+        {
+            const string code1 = @"
+public partial class Sample
+{
+    private static int Original = 42;
+    private int Field = Original;
+}";
+            const string code2 = @"
+public partial class Sample
+{
+    public int Method()
+    {
+        return Field;
+    }
+}";
+            var compilation1 = SolutionBuilder.Create().AddProject(AnalyzerLanguage.CSharp, createExtraEmptyFile: false)
+                .AddSnippet(code1)
+                .GetCompilation();
+            var compilation2 = SolutionBuilder.Create().AddProject(AnalyzerLanguage.CSharp, createExtraEmptyFile: false)
+                .AddSnippet(code2)
+                .GetCompilation();
+            var tree = compilation2.SyntaxTrees.Single(x => x.GetRoot().DescendantNodes().OfType<MethodDeclarationSyntax>().Any());
```

Comment:
I suggest to improve the naming a bit, to make the differences obvious and easier to follow:
- compilation1  -> firstCompilation
- compilation2 -> secondCompilation`

You don't need `tree` from what I see, you can merge that into the LINQ query for returnExpression
Also:
-  returnExpression -> secondCompilationReturnExpression

And then you have:
```
firstCompilationFinder.FindConstant(secondCompilationReturnExpression).Should().BeNull();
```

which is much clearer